management/src/test/java/org/ehcache/management/cluster/ClusteringManagementServiceTest.java

```diff
@@ -0,0 +1,276 @@
+
+public class ClusteringManagementServiceTest {
+
+  static IMonitoringConsumer consumer;
+  static PassthroughClusterControl stripeControl;
+
+  CacheManager cacheManager;
+  ManagementRegistry managementRegistry;
+
+  @Test(timeout = 10000)
+  public void test_programmatic_api() throws Exception {
+    MyService myService = new MyService();
+
+    // like the XML, but programmatic
+
+    cacheManager = CacheManagerBuilder.newCacheManagerBuilder()
+        // cluster config
+        .with(ClusteringServiceConfigurationBuilder.cluster(URI.create("passthrough://server-1:9510/my-server-entity-1"))
+            .autoCreate()
+            .defaultServerResource("primary-server-resource"))
+        .using(myService)
+        // management config
+        .using(new DefaultManagementRegistryConfiguration()
+            .addTags("webapp-1", "server-node-1")
+            .setCacheManagerAlias("my-super-cache-manager")
+            .addConfiguration(new EhcacheStatisticsProviderConfiguration(
+                1, TimeUnit.MINUTES,
+                100, 1, TimeUnit.SECONDS,
+                2, TimeUnit.SECONDS))) // TTD reduce to 2 seconds so that the stat collector run faster
+        // cache config
+        .withCache("cache-1", CacheConfigurationBuilder.newCacheConfigurationBuilder(
+            String.class, String.class,
+            newResourcePoolsBuilder()
+                .heap(10, EntryUnit.ENTRIES)
+                .offheap(1, MemoryUnit.MB)
+                .with(ClusteredResourcePoolBuilder.clusteredDedicated("primary-server-resource", 1, MemoryUnit.MB)))
+            .build())
+        .build(true);
+
+    managementRegistry = myService.managementRegistry;
+
+    runTest();
+  }
+
+  @Test(timeout = 10000)
+  public void test_xml_api() throws Exception {
+
+    cacheManager = CacheManagerBuilder.newCacheManager(new XmlConfiguration(getClass().getResource("/ehcache-management-clustered.xml")));
+    cacheManager.init();
+
+    // At this moment, there is no mechanism in Ehcache to retrieve a configured service from XML at runtime,
+    // except by developing a custom service plus xml parser that will depend on the registry and expose it statically.
+    // But it's not a right solution.
+    Field f = EhcacheManager.class.getDeclaredField("serviceLocator");
+    f.setAccessible(true);
+    ServiceLocator serviceLocator = (ServiceLocator) f.get(cacheManager);
+
+    managementRegistry = serviceLocator.getService(ManagementRegistryService.class);
+
+    runTest();
+  }
+
+  private void runTest() throws Exception {
+    // get the queues where notifs and stats are put
+    BlockingQueue<Object[]> statistics = consumer.getValueForNode(new String[]{"management", "statistics"}, BlockingQueue.class).get();
+    BlockingQueue<Object[]> notifications = consumer.getValueForNode(new String[]{"management", "notifications"}, BlockingQueue.class).get();
+
+    // Issue a management call to configure the stats to collect
```

Comment:
it's not a "management call", but rather a local (ie on the client) management registry configuration change (or update)